beacon-chain/forkchoice/protoarray/types.go

```diff
@@ -44,15 +44,17 @@ type Node struct {
 	weight         uint64                       // weight of this node.
 	bestChild      uint64                       // bestChild index of this node.
 	bestDescendant uint64                       // bestDescendant of this node.
-	graffiti       [fieldparams.RootLength]byte // graffiti of the block node.
+	optimistic     OptimisticStatus             // optimistic status of this node
 }
 
-// optimisticStore defines a structure that tracks the tips of the fully
-// validated blocks tree.
-type optimisticStore struct {
-	validatedTips map[[32]byte]types.Slot
-	sync.RWMutex
-}
+// enum used as optimistic status of a node
+type OptimisticStatus uint8
+
+const (
+	SYNCING OptimisticStatus = iota // the node is optimistic
```

Comment:
I don't think these need to be exported?

```suggestion
	syncing OptimisticStatus = iota // the node is optimistic
```